### PR TITLE
INTMDB 649 CloudBackupSnapshot export job apkeys profile changes

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

CloudBackupSnapshot export job apkeys profile changes

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

### CODE REVIEW

**Comment by govardhanpagidi** [comment]
Tested with PvtExtension

image

File: cfn-resources/cloud-backup-snapshot-export-job/test/cfn-test-delete-inputs.sh [hunk 1]
```diff
@@ -42,3 +43,35 @@ echo "--------------------------------delete role starts -----------------------
...
+while atlas clusters describe "${clusterName}" --projectId "${projectId}"; do
+        sleep 30
+        if atlas clusters describe "${clusterName}" --projectId "${projectId}"
+        then
+          status=$(atlas clusters describe "${clusterName}" --projectId "${projectId}"  --output=json | jq -r '.stateName')
+        else
+          status="DELETED"
+        fi
+        echo "status: ${status}"
+done
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-job/test/cfn-test-delete-inputs.sh:70]: unfortunately, `atlas cluster describe` will return a 404 error once the cluster is deleted. There is no `DELETED` status

File: cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-restore-job.sample-cfn-request.json [hunk 2]
```diff
@@ -2,10 +2,7 @@
     "desiredResourceState": {
         "GroupId": "",
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-restore-job.sample-cfn-request.json:3]:
```suggestion
        "ProjectId": "",
```

File: cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-snapshot-export-job.create-sample-cfn-request.sh [hunk 3]
```diff
@@ -19,12 +20,11 @@ ClusterName="${2}"
...
-	--arg ClusterName "$ClusterName" \
-	--arg group_id "$projectId" \
-	--arg SnapshotId "$snapshotId" \
-	--arg ExportBucketId "$ExportBucketId" \
-	'.desiredResourceState.ExportBucketId?|=$ExportBucketId |.desiredResourceState.SnapshotId?|=$SnapshotId | .desiredResourceState.GroupId?|=$group_id | .desiredResourceState.ApiKeys.PublicKey?|=$pubkey | .desiredResourceState.ApiKeys.PrivateKey?|=$pvtkey | .desiredResourceState.ClusterName?|=$ClusterName' \
-	"$(dirname "$0")/cloud-backup-restore-job.sample-cfn-request.json"
+jq --arg profile "$ATLAS_PROFILE" \
+   --arg org "$ATLAS_ORG_ID" \
+   --arg ClusterName "$ClusterName" \
+   --arg group_id "$projectId" \
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-snapshot-export-job.create-sample-cfn-request.sh:26]:
```suggestion
   --arg project_id "$projectId" \
```

File: cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-snapshot-export-job.create-sample-cfn-request.sh [hunk 4]
```diff
@@ -19,12 +20,11 @@ ClusterName="${2}"
...
+   --arg SnapshotId "$snapshotId" \
+    --arg ExportBucketId "$ExportBucketId" \
+   '.desiredResourceState.ExportBucketId?|=$ExportBucketId |.desiredResourceState.SnapshotId?|=$SnapshotId | .desiredResourceState.GroupId?|=$group_id | .desiredResourceState.Profile?|=$profile | .desiredResourceState.ClusterName?|=$ClusterName' \
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-snapshot-export-job.create-sample-cfn-request.sh:29]:
```suggestion
   '.desiredResourceState.ExportBucketId?|=$ExportBucketId |.desiredResourceState.SnapshotId?|=$SnapshotId | .desiredResourceState.ProjectId?|=$project_id | .desiredResourceState.Profile?|=$profile | .desiredResourceState.ClusterName?|=$ClusterName' \
```

File: cfn-resources/cloud-backup-snapshot-export-job/test/cfn-test-delete-inputs.sh (hunk 1 above)
Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-job/test/cfn-test-delete-inputs.sh:70]: @andreaangiolillo  atlas clusters describe command will become false in WHILE loop and comes out. This has been tested before and working I believe. 
Any thoughts on using other commands just like WATCH here...

File: cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-restore-job.sample-cfn-request.json (hunk 2 above)
Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-restore-job.sample-cfn-request.json]: Updated

File: cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-snapshot-export-job.create-sample-cfn-request.sh (hunk 4 above)
Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-job/test/cloud-backup-snapshot-export-job.create-sample-cfn-request.sh]: Updated

File: cfn-resources/cloud-backup-snapshot-export-job/test/cfn-test-delete-inputs.sh (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/cloud-backup-snapshot-export-job/test/cfn-test-delete-inputs.sh:70]: Thanks for the clarification. No, the watch command won't work here 😢

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**